### PR TITLE
Sync post list 1.1 endpoint.

### PR DESCRIPTION
**Pull request by zinigor**

### CODE REVIEW

**Comment by zinigor** [comment]
Master issue: #7026

File: json-endpoints/class.wpcom-json-api-list-posts-v1-1-endpoint.php [hunk 1]
```diff
@@ -311,6 +312,13 @@ function callback( $path = '', $blog_id = 0 ) {
 						$return[$key]->next_page = $this->build_page_handle( $last_post, $query );
 					}
 				}
+
+				if ( defined( 'IS_WPCOM' ) && IS_WPCOM ) {
+					if ( !isset( $return[$key] ) )
```
Comment (jeherve) [inline json-endpoints/class.wpcom-json-api-list-posts-v1-1-endpoint.php]: Should we take that opportunity to clean up that code so it respects the WP coding standards?

File: json-endpoints/class.wpcom-json-api-list-posts-v1-1-endpoint.php [hunk 2]
```diff
@@ -311,6 +312,13 @@ function callback( $path = '', $blog_id = 0 ) {
...
+						$return[$key] = new stdClass;
+					$return[$key]->wpcom = true;
```
Comment (eliorivero) [inline json-endpoints/class.wpcom-json-api-list-posts-v1-1-endpoint.php]: Although it's not a blocker @zinigor the spacing could be fixed in both sides.

**Comment by zinigor** [comment]
Appreciate the comments, but I'd rather sync this now, and then fix spacing on a file that has a common ancestor. We'll get this all fixed!

**Pull request merged**